File: src/model/menu.ts
import { record } from "../util/prompt";
import { ActionType } from "../util/typeUtil";

export const view = async () => {
  console.log("\nExisting option: ");

  Object.values(ActionType).forEach((action, index) => {
    console.log(` ${index}. ${action}`);
  });
  return Number(await record("\nChoose an option: "));
};

export const amountView = async () => {
  return Number(await record("\nEnter an amount: "));
};

File: model/menu.py
from model.type import ActionType

def view():
    print("\nExisting option: ")

    for index, action in enumerate(ActionType):
        print(f" {index}. {action.value}")

    return int( input("\nChoose an option: "))

def amount_view():
    return float( input("\nEnter an amount: "))
